presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -1105,6 +1096,18 @@ protected Type visitNode(Node node, StackableAstVisitorContext<Context> context)
             throw new SemanticException(NOT_SUPPORTED, node, "not yet implemented: " + node.getClass().getName());
         }
 
+        public Type visitGroupingOperation(GroupingOperation node, StackableAstVisitorContext<Context> context)
+        {
+            if (node.getGroupingColumns().size() > MAX_NUMBER_GROUPING_ARGUMENTS) {
```

Comment:
The above query should actually fail at the parsing stage. SQL grammar doesn't allow expressions as arguments to grouping. So I added a test to `TestParser` for the above query. Wasn't sure how to validate the error so I just did a string match on the error message but I think it makes the test too brittle. Let me know if you have a better idea.